platforms/documentation/docs/src/docs/userguide/best-practices/bp_tasks.adoc

```diff
@@ -121,3 +121,66 @@ include::sample[dir="snippets/bestPractices/useCacheabilityAnnotations/groovy",f
 
 === Tags
 `#tasks`, `#caching`
+
+== Don't resolve Configurations before Task Execution
+
+Resolving configuration before task execution phase can lead to incorrect results and slows gradle build performance.
+
+Once configuration is resolved it can't be resolved again, and it can lose some information when it is resolved too early.
+Calling manual resolution via link:{javadocPath}/org/gradle/api/artifacts/Configuration.html#resolve()[resolve] method returns a set of Files that doesn't preserve references to the tasks that produced these files.
```

Comment:
I think the potential for losing dependency info should be a separate paragraph.

Reply:
I separated them, but didn't fall into thorough explanation. Does it look good?
I'd probably stick to short explanations with links if available.
By the way, do you know any good links to other gradle docs on this topic that could fit here?

Reply:
@lkasso ?

Reply:
Since @lkasso approved the PR, maybe we can let it go as is? And correct if necessary later.
If you're ok with the current explanation.